TestMessageStore.test_write_lock_serializes_concurrent_appends is timing-sensitive on slow hardware

## Summary

`tests/test_lcm_core.py::TestMessageStore::test_write_lock_serializes_concurrent_appends` has a 30-second per-thread `t.join(timeout=30)` that is load-sensitive. On a Raspberry Pi 5 (or any constrained CPU), the 8-thread × 50-write storm can exceed 30s under contention, producing a flaky failure that is **not** a regression in the lock contract being tested.

## Reproduction

**Isolated run (RPi 5, fresh process, no other test activity):**
```
$ pytest tests/test_lcm_core.py::TestMessageStore::test_write_lock_serializes_concurrent_appends -q
1 passed in 7.65s    # run 1
1 passed in 4.29s    # run 2
1 passed in 4.86s    # run 3
```
Average ~5.6s — well under the 30s budget.

**Under CPU contention (RPi 5, 4× `yes > /dev/null` in background):**
```
$ (yes > /dev/null &) && (yes > /dev/null &) && (yes > /dev/null &) && (yes > /dev/null &)
$ pytest ...test_write_lock_serializes_concurrent_appends
1 passed in 25.92s   # uses 86% of the 30s timeout
```
One more CPU-pressure increment and this fails.

**In full suite (RPi 5, 114 tests):**
```
1 failed, 113 passed in 298.53s (0:04:58)
FAILED tests/test_lcm_core.py::TestMessageStore::test_write_lock_serializes_concurrent_appends
tests/test_lcm_core.py:2196: in test_write_lock_serializes_concurrent_appends
    assert not t.is_alive(), "worker thread did not finish in time"
E   AssertionError: worker thread did not finish in time
```

**CI on faster runners (this PR's CI run):** 5/5 green including the affected test — the test is functionally correct, just CPU-bound.

## Workload

```python
n_threads              = 8
per_thread_singles     = 25
per_thread_batches     = 5
per_thread_batch_size  = 5
total writes           = 8 × (25 + 5×5) = 400 SQLite inserts under contention
```

The test (line 2170-2210) is verifying that `MessageStore._write_lock` (an `RLock`) actually serializes concurrent writes — a real and important contract. The 30s `t.join(timeout=30)` is a hard ceiling that doesn't account for slow CI runners or shared hardware.

## Suggested fixes (any of these would work)

1. **Bump the join timeout** to `120` or `300` seconds. Lowest-effort, restores headroom for slow runners. Risk: real concurrency regressions now take longer to detect.

2. **Reduce workload** to e.g. `per_thread_singles=10, per_thread_batches=2, per_thread_batch_size=5` (= 160 total). Still verifies the contract, runs in ~2s even on RPi. Risk: weaker test, may not catch rare interleavings.

3. **Mark the test `@pytest.mark.slow`** and configure CI to skip it on arm/slow runners. Risk: reduces coverage on the affected platforms.

4. **Make the timeout load-aware** — derive it from a slow-mode env var (`HERMES_TEST_SLOW_JOBS=1`) and document it in CONTRIBUTING. Cleanest separation of concerns.

5. **Replace wall-clock assertion with a synchronization primitive** — e.g. `concurrent.futures.ThreadPoolExecutor` with `as_completed(timeout=)` and a per-future budget. More invasive.

My preference: **option 1 (timeout bump)** as the lowest-risk change, with **option 4 (env-var gate)** as a follow-up if the bump alone isn't enough on real CI infrastructure.

## Environment

- OS: Debian 12 on Raspberry Pi 5 (kernel 6.12.87+rpt-rpi-2712)
- Python: 3.11/3.12/3.13/3.14 (all reproduced)
- pytest: 8.x
- Test count: 114 in `test_lcm_core.py`
- Failure mode: thread.join timeout exceeded, not a logic error

## Notes

- The test is **not** flaky on typical CI runners (x86_64 with multi-core) — GitHub Actions shows it consistently passing.
- I hit this while developing PR #245 on an RPi and want to make sure the test doesn't become a paper cut for any future RPi-based contributors or self-hosted CI users.
- I deliberately did **not** fix this in PR #245 — it's outside that PR's scope and the maintainer may prefer a different approach than the options above.

--- a/tests/test_lcm_core.py
+++ b/tests/test_lcm_core.py
@@ -2191,9 +2191,17 @@ class TestMessageStore:
         ]
         for t in threads:
             t.start()
+
+        # Slow hardware can make a single worker exceed the old fixed
+        # per-thread 30s join while the write-lock contract is still healthy.
+        # Use one larger wall-clock deadline for the whole storm: enough
+        # headroom for constrained machines, but still bounded for real hangs.
+        deadline = time.monotonic() + 120
         for t in threads:
-            t.join(timeout=30)
-            assert not t.is_alive(), "worker thread did not finish in time"
+            remaining = max(0.0, deadline - time.monotonic())
+            t.join(timeout=remaining)
+        alive_threads = [t.name for t in threads if t.is_alive()]
+        assert alive_threads == [], f"worker threads did not finish in time: {alive_threads!r}"
 
         assert errors == [], f"concurrent workers raised: {errors!r}"
 

